### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Shtong**
Here is a first working draft to address #87. It's getting pretty big, since in addition to editing `.editorconfig` to the rules we talked about, I also started to apply the rules to the code.

A few things to note however:
- I have done my best to not touch anything that would have broken compatibility for the library users. So for example, public async methods were not renamed to have an Async suffix.
- I have sometimes made changes to follow the [official capitalization rules for acronyms](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions?redirectedfrom=MSDN#capitalization-rules-for-identifiers): for example, `conformanceTOCReady` was renamed `_conformanceTocReady` (with lowercase 'oc')
- In some places I took the liberty to replace `ArgumentException` (or an equivalent) with `InvalidOperationException`, which is the correct exception that should be used when the state of an object (and not an argument) is causing a problem.

I'm not sure about the end result to be honest. Because I wanted to not break the public interface, it looks half-baked. Also because of the very large amount of small changes, the PR looks very hard to review. I feel it should either be less ambitious, or go all the way and break the existing API in the process, maybe producing a new major version. Any thoughts ?

### CODE REVIEW

**Comment by aseigler** [comment]
This is actually not too bad to review, as all the code is extremely familiar to me.

As far as the capitalization rules go, in many cases I used the abbreviations directly from the WebAuthn spec, various FIDO specs, or in the case of TPM attestation, TCG specs.  In the specific example you gave, in the FIDO specs the metadata [table of contents](https://fidoalliance.org/specs/fido-v2.0-rd-20180702/fido-metadata-service-v2.0-rd-20180702.html#metadata-toc) is always referred to as a TOC, so I tried my best to match the case from the spec exactly, so that someone trying to map the code in the library to the spec would find it very easy to follow.  If that's not proper, that's fine, we can name things however the cool kids are doing it these days.  So long as it works and makes sense, I am ok with that.

File: Src/Fido2/AttestationFormat/AndroidKey.cs [hunk 1]
```diff
@@ -21,25 +23,34 @@ public static byte[] AttestationExtensionBytes(X509ExtensionCollection exts)
...
+                throw new Fido2VerificationException("attExtBytes signature not a valid DER sequence");
+
             // next is length of all the items in the sequence
             var dataLen = AuthDataHelper.GetSizedByteArray(attExtBytes, ref offset, 1);
-            if (null == dataLen) throw new Fido2VerificationException("attExtBytes signature has invalid length");
+            if (null == dataLen)
+                throw new Fido2VerificationException("attExtBytes signature has invalid length");
+
             // if data is more than 127 bytes, the length is encoded in long form
+            // TODO : Why is longLen never used ?
```
Comment (aseigler) [inline Src/Fido2/AttestationFormat/AndroidKey.cs]: Because I forgot to bring over the sanity check added on that field I added in #100.

**Comment by aseigler** [comment]
LGTM but we introduced some conflicts by pulling #125. Should not be too bad to resolve.

**Comment by Shtong** [comment]
Alright. I started merging; although I want to make a pass on the many new files so they use the same rules as the other ones I already refactored before we finish that PR.

**Comment by aseigler** [comment]
Thank you!

**Pull request merged**